### PR TITLE
Feat/update treesitter

### LINKED ISSUE #25: Neovim crashing when angular and typescript parsers are installed
When ***both*** angular and typescript parsers are installed, and I delete a specific line, in a special file, neovim crashes.
Here is the recording [asciinema](https://asciinema.org/a/d1EYiTv2Uke7xhE0jPgHVYJ8x)
## Note:
Neovim crashes when I press `dd` on the innermost line.

Here is the file:
```typescript
@Component({
  template: `
    <section >
      <router-outlet> </router-outlet>
    </section>
`,
})
```
Here is a minimal neovim config to reproduce the behaviour:
```lua
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({
		"git",
		"clone",
		"--filter=blob:none",
		"https://github.com/folke/lazy.nvim.git",
		"--branch=stable", -- latest stable release
		lazypath,
	})
end
vim.opt.rtp:prepend(lazypath)

-- require("options") -- Important to go first
require("lazy").setup({
{
	"nvim-treesitter/nvim-treesitter",
	build = ":TSUpdate",
	config = function()
		local configs = require("nvim-treesitter.configs")

		configs.setup({
			sync_install = true,
			highlight = { enable = true },
			indent = { enable = false },
			incremental_selection = {
				enable = true,
				keymaps = {
					init_selection = "<CR>", -- set to `false` to disable one of the mappings
					node_incremental = "<CR>",
					scope_incremental = "<TAB>",
					node_decremental = "<S-TAB>",
				},
			},
		})
	end,
}
})

```

### PR DESCRIPTION
**Pull request by dlvandenberg**

### CODE REVIEW

**Comment by iurimateus** [comment]
Thanks. I've been using this PR for a while, no crashes.

File: package.json [hunk 1]
```diff
@@ -32,11 +43,21 @@
...
+  },
+  "peerDependencies": {
+    "tree-sitter": "^0.21.0"
+  },
+  "peerDependenciesMeta": {
+    "tree_sitter": {
+      "optional": true
+    }
   },
   "tree-sitter": {
```
Comment (clason) [inline package.json]: This is actually incorrect (and thus breaks `:TSInstallFromGrammar`); it needs to be an array: `"tree-sitter": [ {...} ]`.

Comment (dlvandenberg) [reply package.json]: Ah, thanks for catching this one

**Comment by clason** [comment]
`tree-sitter generate` (with CLI 0.22.5) will bump a lot of this for you ;)

File: package.json [hunk 2]
```diff
@@ -43,16 +43,16 @@
     "@stylistic/eslint-plugin-js": "^1.4.1",
     "eslint": "^8.54.0",
     "eslint-config-google": "^0.14.0",
-    "tree-sitter-cli": "^0.22.0",
+    "tree-sitter-cli": "^0.22.2",
```
Comment (clason) [inline package.json]:
```suggestion
    "tree-sitter-cli": "^0.22.5",
```
is the latest

Comment (dlvandenberg) [reply package.json]: Thanks :)

**Comment by dlvandenberg** [comment]
Still a failing npm install on windows. Any ideas @clason ? I'm out of ideas..

**Comment by clason** [comment]
I'd try out https://github.com/tree-sitter/workflows, to be honest.

But first thoughts:
1. bump node to v20
2. let `tree-sitter generate` the bindings for you and test those.

(I don't use Windows myself so can't be of much help there specifically.)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -2,36 +2,72 @@ name: CI
...
-    branches:
-      - main
+    branches: [main]
+    path:
+      - grammar.js
+      - src/**
+      - test/**
+      - bindings/**
+      - binding.gyp
+
```
Comment (clason) [inline .github/workflows/ci.yml:12]:
```suggestion
```

**Comment by dlvandenberg** [comment]
Should fix #25

**Comment by gipo355** [comment]
Fixed #25 for me

**Pull request merged**